packages/integrations/vercel/src/static/adapter.ts

```diff
@@ -118,6 +118,11 @@ export default function vercelStatic({
 							continue: true,
 						},
 						{ handle: 'filesystem' },
+						...routes.find(route => route.component.endsWith("/pages/404.astro")) ? [{
```

Comment:
Not blocking at all: is `route.component` the right thing to check here? Seems like `route.pathname?.endsWith('/404')` might be better?

Maybe we should expose if the site has a custom 404 page to adapters...

Reply:
Yeah we definitely should. I would've been the happiest with an exact check for `src/pages/404.astro`, but `src` is configurable so this is the next best option. Do you know off the top of your head whether `pathname` is immune from configuration options, like `trailingSlash`?